### PR TITLE
greenhills support: add new platform of qemu cortex-m7

### PR DESCRIPTION
**Pull request by extinguish**
## Summary
1. Support NuttX running on qemu cortex-m7, and greenhills can be verified on this platform

### CODE REVIEW

File: boards/arm/mps/mps2-an500/configs/nsh/defconfig
Comment (xuxin930) [inline boards/arm/mps/mps2-an500/configs/nsh/defconfig:1]: please refresh this defconfig to solve ci break.

Comment (extinguish) [reply boards/arm/mps/mps2-an500/configs/nsh/defconfig:1]: > please refresh this defconfig to solve ci break.

very grateful thanks

**Review by anchao — APPROVED** [review]

File: boards/Kconfig [hunk 1]
```diff
@@ -2119,6 +2119,14 @@ config ARCH_BOARD_MR_CANHUBK3
 		This options selects support for NuttX on the NXP MR-CANHUBK3 board
 		featuring the S32K344 Cortex-M7.
 
+config ARCH_BOARD_MPS2_AN500
+	bool "Arm MPS2 AN500"
+	depends on ARCH_CHIP_MPS2_AN500
```
Comment (xuxin930) [inline boards/Kconfig:2124]: where this config defined? 
is this should define in arch Kconfig?

Comment (xuxin930) [reply boards/Kconfig:2124]: i mean depends on `ARCH_CHIP_MPS2_AN500`

**Comment by extinguish** [comment]
All comments are fixed, please help to make a review on this patch, very grateful thanks

**Comment by extinguish** [comment]
@xuxin930 all comments are fixed, please help to take a review, very grateful thanks

**Review by pkarashchenko — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**